src/onboarding/onboarding-restore-backup.ts

```diff
@@ -1,136 +1,328 @@
+    this._loadBackupInfo();
   }
 
-  private _backupCleared() {
-    this._backupSlug = undefined;
-  }
+  private async _loadBackupInfo() {
+    const backupId = localStorage.getItem(STORAGE_BACKUP_ID_KEY);
+    const restoreRunning = localStorage.getItem(STORAGE_RESTORE_RUNNING);
 
-  protected firstUpdated(changedProps) {
-    super.firstUpdated(changedProps);
-  }
+    try {
+      const {
+        last_non_idle_event: lastNonIdleEvent,
+        state: currentState,
+        backups,
+      } = await fetchBackupOnboardingInfo();
+
+      this._backupInfo = {
+        state: currentState,
+        last_non_idle_event: lastNonIdleEvent,
+      };
 
-  private async _checkRestoreStatus(): Promise<void> {
-    if (this._restoring) {
-      try {
-        await fetchInstallationType();
-      } catch (err: any) {
+      if (backupId) {
+        this._backup = backups.find(({ backup_id }) => backup_id === backupId);
+      }
+
+      const failedRestore =
+        lastNonIdleEvent?.manager_state === "restore_backup" &&
+        lastNonIdleEvent?.state === "failed";
+
+      if (failedRestore) {
+        this._failed = true;
+      }
+
+      if (restoreRunning) {
+        this._view = "status";
+        if (failedRestore || currentState !== "restore_backup") {
+          this._failed = true;
+          localStorage.removeItem(STORAGE_RESTORE_RUNNING);
+        } else {
+          this._scheduleLoadBackupInfo();
+        }
+        return;
+      }
+
+      if (!lastNonIdleEvent || !backupId) {
+        this._view = "upload";
+        return;
+      }
+
+      if (
+        this._backup &&
+        (lastNonIdleEvent.manager_state === "receive_backup" || failedRestore)
+      ) {
+        if (!this.supervisor && this._backup.homeassistant_included) {
+          this._selectedData = {
+            homeassistant_included: true,
+            folders: [],
+            addons: [],
+            homeassistant_version: this._backup.homeassistant_version,
+            database_included: this._backup.database_included,
+          };
+          this._view = "restore";
+        } else {
+          this._view = "details";
+        }
+        return;
+      }
+
+      // fallback to upload
+      this._view = "upload";
+    } catch (err: any) {
+      if (restoreRunning) {
         if (
-          (err as Error).message === "unauthorized" ||
-          (err as Error).message === "not_found"
+          err.error === "Request error" ||
+          // core can restart but haven't loaded the backup integration yet
+          (err.status_code === 500 && err.body?.error === "backup_disabled")
         ) {
-          window.location.replace("/");
+          this._scheduleLoadBackupInfo();
+          return;
+        }
+
+        // core seems to be back up restored
+        if (err.status_code === 404) {
+          localStorage.removeItem(STORAGE_RESTORE_RUNNING);
+          localStorage.removeItem(STORAGE_BACKUP_ID_KEY);
+          location.reload();
```

Comment:
Shouldnt we navigate to the root?
```suggestion
          window.location.replace("/");
```